apps/elasticsearch.go

```diff
@@ -5,8 +5,99 @@ import (
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
 )
 
+type MetricsReceiverElasticsearch struct {
+	confgenerator.ConfigComponent          `yaml:",inline"`
+	confgenerator.MetricsReceiverShared    `yaml:",inline"`
+	confgenerator.MetricsReceiverSharedTLS `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url,startswith=http:|startswith=https:"`
+
+	Username string `yaml:"username" validate:"required_with=Password"`
+	Password string `yaml:"password" validate:"required_with=Username"`
+
+	CollectJVMMetrics  *bool `yaml:"collect_jvm_metrics" validate:"omitempty"`
+	SkipClusterMetrics *bool `yaml:"skip_cluster_metrics" validate:"omitempty"`
+}
+
+const (
+	defaultElasticsearchEndpoint           = "http://localhost:9200"
+	defaultElasticsearchSkipClusterMetrics = true
+)
+
+func (r MetricsReceiverElasticsearch) Type() string {
+	return "elasticsearch"
+}
+
+func (r MetricsReceiverElasticsearch) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		r.Endpoint = defaultElasticsearchEndpoint
+	}
+
+	cfg := map[string]interface{}{
+		"collection_interval": r.CollectionIntervalString(),
+		"endpoint":            r.Endpoint,
+		"username":            r.Username,
+		"password":            r.Password,
+		"nodes":               []string{"_local"},
+		"tls":                 r.TLSConfig(true),
+	}
+
+	if r.CollectJVMMetrics != nil && !*r.CollectJVMMetrics {
```

Comment:
Can we refactor this part a bit to use the common util by https://github.com/GoogleCloudPlatform/ops-agent/pull/370?

Reply:
I don't think it's correct to use that here.

To explain further; The Elasticsearch metrics receiver is not JMX based; It uses an http endpoint, where Elasticsearch exposes JVM information. That common util is for JMX receivers, and thus isn't suitable for this case.

Reply:
Oh I see. Thanks for the clarification.